### PR TITLE
Add empty plugins dir for archive distributions

### LINKED ISSUE #20342: elsticsearch-plugin list results in error message if there are no plugins installed
**Elasticsearch version**: latest master (2b43c6db3a5bdd822e14b091925bae609b4e286e)

**Plugins installed**: []

**Description of the problem including expected versus actual behavior**: When I run 

`elsticsearch-plugin list` 

with no plugin installed I would expect the list returned is empty. Instead I get:

```
$ ./elasticsearch-5.0.0-alpha6-SNAPSHOT/bin/elasticsearch-plugin  list
Exception in thread "main" java.io.IOException: Plugins directory missing: /home/britta/es-token-plugin/src/test/resources/r-demo/elasticsearch-5.0.0-alpha6-SNAPSHOT/plugins
    at org.elasticsearch.plugins.ListPluginsCommand.execute(ListPluginsCommand.java:51)
    at org.elasticsearch.cli.SettingCommand.execute(SettingCommand.java:54)
    at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:88)
    at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:69)
    at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:88)
    at org.elasticsearch.cli.Command.main(Command.java:54)
    at org.elasticsearch.plugins.PluginCli.main(PluginCli.java:57)

```

### PR DESCRIPTION
**Pull request by jasontedor**
Today when installing Elasticsearch from an archive distribution (tar.gz
or zip), an empty plugins folder is not included. This means that if you
install Elasticsearch and immediately run elasticsearch-plugin list, you
will receive an error message about the plugins directory missing. While
the plugins directory would be created when starting Elasticsearch for
the first time, it would be better to just include an empty plugins
directory in the archive distributions. This commit makes this the
case. Note that the package distributions already include an empty
plugins folder.

Closes #20342

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]
Looks right. Can you add a test that runs the plugin command without any plugins installed?

**Comment by jasontedor** [comment]
@nik9000 I pushed c411dd6d6806aa0d4a01c40e199882c5c34d194f.

**Comment by jasontedor** [comment]
test this please

**Comment by jasontedor** [comment]
@nik9000 Are you good with the test that was pushed in c411dd6d6806aa0d4a01c40e199882c5c34d194f.

File: qa/vagrant/src/test/resources/packaging/scripts/20_tar_package.bats [hunk 1]
```diff
@@ -73,6 +73,13 @@ setup() {
     verify_archive_installation
 }
 
+@test "[TAR] verify elasticsearch-plugin list runs" {
```
Comment (nik9000) [inline qa/vagrant/src/test/resources/packaging/scripts/20_tar_package.bats]: s/runs/runs without any plugins installed/

**Review by nik9000 — APPROVED** [review]

**Comment by nik9000** [comment]
Tests look good I'd change the name of the test case so it is super obvious.

**Comment by jasontedor** [comment]
> Tests look good I'd change the name of the test case so it is super obvious.

Sure, I pushed e42513c1a0e4521929e33a019068979ad5dadda0. Note that the test does assert that the output from `elasticsearch-plugin list` is empty to ensure that the test is not moved somewhere after some plugins are installed.

**Pull request merged**